Added a TitleChanged event handler

perlun opened the pull request:
This one adds a `TitleChanged` event handler, which is needed in my use case (so that I don't have to do dirty hacks).

Sorry for the whitespace changes; they were included without without extra charge. :wink: 

OK to merge?

perlun commented on CefSharp.Wpf/ChromiumWebBrowser.cs:
```diff
@@ -524,7 +523,21 @@ public string Title
+            {
+                handlers(owner, e);
+            }
+        }
```
This is the essence of the PR.

perlun commented:
/cc @jornh also since you are in a more correct time zone than Alex. :smile:

amaitland commented:
@perlun This mean your finally upgrading to `CefSharp3`? :smile:

perlun commented:
> @perlun This mean your finally upgrading to CefSharp3? :smile:

Yes! :smile: Finally...

I will make a 37.0.1 hotfix to get this fix available. No expert on AppVeyor - do you just press the Deploy button or how do you do it?

perlun commented:
Btw - feel free to adjust as you see fit. You run the project now, so do whatever you feel like. I gave my comments above but feel free to accept or reject them, or fine-tune the patch to your liking.

perlun commented:
I noted that my build failed now, for whatever reason: https://ci.appveyor.com/project/cefsharp/cefsharp/build/37.0.0-CI774#L33

@amaitland or @jornh, if you could give me a short explanation on how to build & publish a NuGet package these days I would be extremely delighted! :smile: I guess I could just compile it locally & push, but I cannot even get that working:

```
"Y:\git\CefSharp-37\CefSharp3.sln" (rebuild target) (1) ->
"Y:\git\CefSharp-37\CefSharp.Test\CefSharp.Test.csproj" (Rebuild target) (2) ->
"Y:\git\CefSharp-37\CefSharp.BrowserSubprocess\CefSharp.BrowserSubprocess.csproj" (default target) (3:2) ->
"Y:\git\CefSharp-37\CefSharp.BrowserSubprocess.Core\CefSharp.BrowserSubprocess.Core.vcxproj" (default target) (4:2) ->
"Y:\git\CefSharp-37\CefSharp.Core\CefSharp.Core.vcxproj" (default target) (6:5) ->
(Link target) ->
  LINK : fatal error LNK1181: cannot open input file 'opengl32.lib' [Y:\git\CefSharp-37\CefSharp.Core\CefSharp.Core.vcxproj]
```

amaitland commented:
`AppVeyor` made some changes to their environment, so I had to modify the build script subsequent to `37`'s release. I think the relevant commit is https://github.com/cefsharp/CefSharp/commit/72270b155b82515c86edb51faff0ae377ce92f95#diff-f0e02e42e53ec40f5d3b8bed031ad09e

You'll likely have to delete the `tag` you created and recreate, as the script only pushes official release builds when it detects a `tag` (they'll be tagged with the `CI` prefix otherwise)

I have to rush out, will provide more detail if require when I get home. Hope this helps!

amaitland commented:
Should also note that packages are pushed to `MyGet`, which can then be pushed to `nuget` using the web interface.

Do you have a `MyGet` account?

perlun commented:
> Do you have a MyGet account?

Now I have - it's `perlun`. Care to add me to the CefSharp team?

perlun commented:
The build was successful now, so now I "just" need to push the package to NuGet.org. Ping @amaitland and/or @jornh.

amaitland commented:
> Now I have - it's perlun. Care to add me to the CefSharp team?

Done.

amaitland commented:
Also updated the build script to increment the version number see https://github.com/cefsharp/CefSharp/commit/df5d8f82b96a9668ee4bcca7ceb792f2cca22c56

I removed the previous `37.0.1` tag and created a new release. Hopefully the `nuget` packages will appear on `MyGet` in the next 10mins or so.

amaitland commented:
> Btw - feel free to adjust as you see fit. You run the project now, so do whatever you feel like. I gave my comments above but feel free to accept or reject them, or fine-tune the patch to your liking.

I'd prefer a democracy :smile: Will leave the code as is for now, will have a ponder about a forward direction. Maybe we need a better defined coding standard overall? More than just a `follow what's there`?

amaitland commented:
@perlun Packages are up on `MyGet`. https://www.myget.org/feed/Packages/cefsharp

Mind taking them for a spin first before pushing to `NuGet`? If there are any problems, it's easy to delete the packages from `MyGet`, delete the `GitHub` tag/release and publish a new set. (No need to increment the build version). 

When your ready, you can push the four packages to `Nuget`. Usually you'd just use `Push Latest` and they'd get uploaded them in bulk. As they're semantically at the bottom of the list, we'll have to push the four packages individually.

perlun commented:
Great, thanks! Btw, seems to be a speling error in build.ps1 (assemly)

Sent from my iPhone

On 17 mar 2015, at 14:42, Alex Maitland <notifications@github.com<mailto:notifications@github.com>> wrote:


Also updated the build script to increment the version number see df5d8f8<https://github.com/cefsharp/CefSharp/commit/df5d8f82b96a9668ee4bcca7ceb792f2cca22c56>

I removed the previous 37.0.1 tag and created a new release. Hopefully the nuget packages will appear on MyGet in the next 10mins or so.

-
Reply to this email directly or view it on GitHub<https://github.com/cefsharp/CefSharp/pull/885#issuecomment-82324331>.

amaitland commented:
> Btw, seems to be a speling error in build.ps1 (assemly)

Was fixed in `master` a while ago :smile: 

Can you access `MyGet` now?

perlun commented:
@amaitland - yes, thanks, I now have access. I want to push this package now (using it myself and "seems to work", kind of). How do I do it on MyGet? I feel like a n00b. :smile: 

Also - I noticed that 39.0.0 has been replaced. How breaking is it compared to 37.0.0? Just about to release a version of my stuff based on 37.0.1, but wanted to check if 39.0.0 is worthwhile... What are the major Chromium changes in that version btw?

amaitland commented:
> How do I do it on MyGet?

You'll have to select each package individually, just press `Push`, a popup opens, you just press `Push` again. It's all linked to directly upload to `Nuget.org`, usually takes 2-3mins to appear.

> I noticed that 39.0.0 has been replaced. How breaking is it compared to 37.0.0?

Breaking changes are in the release notes see https://github.com/cefsharp/CefSharp/releases/tag/v39.0.0 A few things were renamed, couple of params added, all fairly trivial stuff.

> What are the major Chromium changes in that version btw?

- GPU Accelerated `OSR` rendering
- Native `PDF` support
- `OSR` printing works again
- Spellchecking support
- Basic `GeoLocation` support

rassilon commented:
Anything after http://googlechromereleases.blogspot.com/2014/08/stable-channel-update_26.html up to http://googlechromereleases.blogspot.com/2014/12/stable-channel-update.html in the stable channel update posts for Windows. CefSharp's Changelog covers the CefSharp specific stuff.

The breaking changes in CefSharp are mostly just new parameter or class/interface renamings. If you're happy now, you should still be happy. :smile: 

I also managed to sneak a major fixing of our WinForm focus problems into 39.0.0 (focus take 5) which is pretty sweet. There was also some commits to try and improve HiDPI with WPF.

Fyi,
Bill
